test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -999,10 +1003,54 @@ public async void DisposeAsync_Should_Dispose_Stream_Asynchronously_AfterWriting
 
             Assert.True(stream.Disposed);
         }
+
+        [Fact]
+        public void WriteMetadataDocument_WorksForLegacyScaleAndSridVariables()
+        {
+            // Arrange
+            IEdmModel edmModel = GetScaleAndSridModel();
+
+            // Act
+            string payload = this.WriteAndGetPayloadForScaleAndSridVariables(edmModel, useLegacyScaleAndSridVariable : true, omWriter =>
+            {
+                omWriter.WriteMetadataDocument();
+            });
+
+            // Assert
+            Assert.Equal("<?xml version=\"1.0\" encoding=\"utf-8\"?>" +
+                "<edmx:Edmx Version=\"4.0\" xmlns:edmx=\"http://docs.oasis-open.org/odata/ns/edmx\">" +
+                "<edmx:DataServices><Schema Namespace=\"NS\" xmlns=\"http://docs.oasis-open.org/odata/ns/edm\">" +
+                "<ComplexType Name=\"Complex\"><Property Name=\"GeographyPoint\" Type=\"Edm.GeographyPoint\" SRID=\"Variable\" />" +
+                "<Property Name=\"DecimalProperty\" Type=\"Edm.Decimal\" Nullable=\"false\" Scale=\"Variable\" />" +
+                "</ComplexType>" +
+                "</Schema></edmx:DataServices></edmx:Edmx>", payload);
+        }
+
+        [Fact]
+        public void WriteMetadataDocument_WorksFor_LowerScaleAndSridVariable()
+        {
+            // Arrange
+            IEdmModel edmModel = GetScaleAndSridModel();
+
+            // Act
+            string payload = this.WriteAndGetPayloadForScaleAndSridVariables(edmModel, useLegacyScaleAndSridVariable : false, omWriter =>

```

Comment:
```suggestion
            string payload = this.WriteAndGetPayloadForScaleAndSridVariables(edmModel, useLegacyScaleAndSridVariable : false, messageWriter =>
```
Because `messageWriter` is obviously a better variable name than `omWriter`